### PR TITLE
Feat/ST-327 applicant: Enter applicant details

### PR DESCRIPTION
**Pull request by sjamihmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ST-327

### Change description ###
Add page to enter applicant details

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by muradalamv1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java [hunk 1]
```diff
@@ -127,8 +147,13 @@ public class CicCase {
         label = "What is their contact preference?",
         access = {DefaultAccess.class, CaseworkerWithCAAAccess.class}
     )
-    private ContactPreferenceType contactDetailsPreference;
+    private ContactPreferenceType contactDetailsPrefrence;
     private ApplicantCICDetails applicantCICDetails;
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java:151]: Don't unfix this typo.

Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java:151]: Please don't unfix the typo

**Review by osheaf01 — CHANGES_REQUESTED** [review]
Don't unfix the typos, otherwise LGTM

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDocumentsCIC.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    private static <T> List<ListValue<T>> sortDocuments(final Map<Boolean, List<ListValue<T>>> documentsWithoutIds) {
+        final List<ListValue<T>> sortedDocuments = new ArrayList<>();
+        final var newDocuments = documentsWithoutIds.get(true);
+        final var previousDocuments = documentsWithoutIds.getOrDefault(false, new ArrayList<>());
+
+        if (null != newDocuments) {
+            sortedDocuments.addAll(0, newDocuments); // add new documents to start of the list
+            sortedDocuments.addAll(1, previousDocuments);
+            sortedDocuments.forEach(
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDocumentsCIC.java:83]: Mentioned this on Murad's PR. Can understand adding something to 0 - the start - but not at 1 - which presumes the previous add had a size of 1?
If you want to order it previousDocuments, newDocuments, then do:
```
sortedDocuments.addAll(0, previousDocuments);
sortedDocuments.addAll(0, newDocuments);   
``` 
That is, reverse the order?

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java [hunk 3]
```diff
@@ -127,8 +129,78 @@ public class CicCase {
...
+    )
+    private String applicantFullName;
+
+    @CCD(label = "Address")
+    private AddressGlobalUK applicantAddress;
+
+    @CCD(
+        label = "Phone number",
+        regex = "^[0-9 +().-]{9,}$"
+    )
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java:146]: Can we test this regex at some point, with a unit test?

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java [hunk 4]
```diff
@@ -127,8 +129,78 @@ public class CicCase {
...
+    private String representativeFullName;
+
+    @CCD(label = "Address")
+    private AddressGlobalUK representativeAddress;
+
+    @CCD(
+        label = "Phone number",
+        regex = "^[0-9 +().-]{9,}$"
+    )
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java:180]: As above, should write a unit test for this regex

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDocumentsCIC.java (hunk 2 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDocumentsCIC.java:83]: Think this is already handled as part of Murad's PR.

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java (hunk 3 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java:146]: @osheaf01 , As discussed on standup, it is not possible to test regex using unit tests as the validation happens by CCD component.

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java (hunk 4 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java:192]: @osheaf01 , As discussed on standup, it is not possible to test regex using unit tests as the validation happens by CCD component.

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java (hunk 1 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java]: Done.

Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CicCase.java]: Done

**Review by jaberelfasi — DISMISSED** [review]

**Review by muradalamv1 — APPROVED** [review]

**Pull request merged**